File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSWhiteboardExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.whiteboard.HMSWhiteboard
import live.hms.video.whiteboard.State

class HMSWhiteboardExtension {
    companion object {
        fun toDictionary(hmsWhiteboard: HMSWhiteboard?): HashMap<String, Any?>? {
            if (hmsWhiteboard == null) {
                return null
            }

            val whiteboardMap = HashMap<String, Any?>()

            whiteboardMap["id"] = hmsWhiteboard.id
            whiteboardMap["owner"] = HMSPeerExtension.toDictionary(hmsWhiteboard.owner)
            whiteboardMap["title"] = hmsWhiteboard.title
            whiteboardMap["url"] = hmsWhiteboard.url
            whiteboardMap["state"] = getStateFromString(hmsWhiteboard.state)
            whiteboardMap["is_owner"] = hmsWhiteboard.isOwner
            return whiteboardMap
        }

        private fun getStateFromString(state: State): String {
            return when (state) {
                State.Started -> "started"
                State.Stopped -> "stopped"
                else -> "stopped"
            }
        }
    }
}

File: packages/hmssdk_flutter/ios/Classes/Models/HMSWhiteboardExtension.swift
//
//  HMSWhiteboardExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 30/04/24.
//

import Foundation
import HMSSDK

class HMSWhiteboardExtension {

    static func toDictionary(hmsWhiteboard: HMSWhiteboard?, hmsSDK: HMSSDK?) -> [String: Any?]? {

        guard let whiteboard = hmsWhiteboard else {
            return nil
        }

        var args = [String: Any?]()

        args["id"] = hmsWhiteboard?.id

        if let owner = hmsWhiteboard?.owner {
            args["owner"] = HMSPeerExtension.toDictionary(owner)
        }
        args["title"] = hmsWhiteboard?.title
        args["url"] = hmsWhiteboard?.url?.absoluteString
        args["is_owner"] = hmsWhiteboard?.owner?.customerUserID == hmsSDK?.localPeer?.customerUserID
        args["state"] = getStateFromString(state: hmsWhiteboard?.state)

        return args
    }

    private static func getStateFromString(state: HMSWhiteboard.WhiteboardState?) -> String {
        switch state {
        case .started:
            return "started"
        case .stopped:
            return "stopped"
        default:
            return "stopped"
        }
    }
}
